x-pack/plugin/autoscaling/src/test/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveReasonTests.java

```diff
@@ -77,6 +105,55 @@ public void testXContent() throws IOException {
             );
             assertSorted(xContentAssignedShardIds.stream().map(ShardId::fromString).toList());
             assertEquals(assignedShardIds.size(), map.get("assigned_shards_count"));
+
+            Map<String, Object> unassignedNodeDecisions = (Map<String, Object>) ((Map<String, Object>) map.get("unassigned_node_decisions"))
+                .get(unassignedShardIds.first().toString());
+            for (Map<String, Object> canAllocateDecision : (List<Map<String, Object>>) unassignedNodeDecisions.get(
+                "can_allocate_decisions"
+            )) {
+                assertEquals("node1", canAllocateDecision.get("node_id"));
```

Comment:
Thanks, the validation in this test case makes it easy to reason about the structure of the result.